pkg/logs/input/file/scanner.go

```diff
@@ -69,6 +68,7 @@ func NewScanner(sources *config.LogSources, tailingLimit int, pipelineProvider p
 		tailerSleepDuration:    tailerSleepDuration,
 		stop:                   make(chan struct{}),
 		validatePodContainerID: validatePodContainerID,
+		scanPeriod:             time.Duration(coreConfig.Datadog.GetFloat64("logs_config.file_scan_period") * float64(time.Second)),
```

Comment:
for consistency with how other options are passed, can you pass the option's value to `NewScanner` instead?